### DISCUSSION TITLE
Install TRMM Agent on a synology DSM 7

### DISCUSSION THREAD

**Discussion by LSIParis** [amidaware/tacticalrmm#1172] [Q&A]
I have a bunch of Syno I'd like to integrate in TRMM
As far as I know it should possiblel but I never succeded

Which versoin of the agent do I need to download ?

Thanks for your help

BTW : What's about QNAP ?

**Comment by dinger1986** [comment]
yes it will work, just need to install the linux agent

**Reply by wh1te909** [reply]
yea for synology, linux should work. for QNAP you probably need to compile using `GOARCH=arm` and maybe even `GOARM=5` please lookup how to cross compile golang for different operating systems and then adjust the build command in the agent repo

**Reply by LSIParis** [reply]
I'll give it a look

**Comment by LSIParis** [comment]
thanks for all your reply
I tried a 2 diiferents syno and I have this error from systemctl : 
systemctl: unrecognized option --now
![PicServ1706](https://user-images.githubusercontent.com/49915017/174314714-dc9a2f3d-5d15-4544-958c-ba2bceafa715.png)


Any clue

**Comment by wh1te909** [comment]
what is the exact model of the synology? you might need to recompile for a different architecture (exec format error usually means wrong arch) and edit the installation script to use synology's service manager instead of systemd which doesn't sound like your synology supports

**Comment by herzkerl** [comment]
Ran into the same issue on a [RS422+](https://www.synology.com/de-de/products/RS422+)—are there any news regarding out-of-the-box Synology support?

```bash
user@NAS:~$ sudo ./script.sh
Password:
Downloading tactical agent...
Downloading and installing mesh agent...
env: ‘/tmp/mesh-T6viH6040/meshagent’: Permission denied
Getting mesh node id...
INFO[0001] Adding agent to dashboard
ERRO[0010] GetWMIInfo() ps.Host() 1 error: no valid /etc/<distrib>-release file found: 0 errors:
INFO[0011] Installation was successfull!
Allow a few minutes for the agent to properly display in the RMM
systemctl: unrecognized option '--now'
user@NAS:~$ client_loop: send disconnect: Broken pipe
```

**Reply by dinger1986** [reply]
I'll try it on afew boxes.

See if I can see what is wrong

**Reply by dinger1986** [reply]
looked at the synology, theres 2 issues, first of all the `systemctl enable/disable --now servicename` seems to not be available on synology, so it could maybe be replaced with `systemctl enable/disable servicename` and `systemctl start/stop servicename` 

Then the issue of the release file not existing, seems to mean its perpetually stuck on 53 years but everything else works

**Reply by wh1te909** [reply]
thanks @dinger1986 I've updated the linux install script in https://github.com/amidaware/tacticalrmm/commit/8a2eb7b058d393e2dd15c6794242b046b6e21ca5 to use the older `systemd` syntax since `--now` was added in later versions. for the `/tmp` permission denied error that should also be fixed when I do the next release shortly, that was fixed in a recent commit see https://github.com/amidaware/rmmagent/issues/32 and will be released with agent v2.4.5

**Comment by LSIParis** [comment]
Hi there,

Sorry no feedback, I gave up !
Most of my customers are switching to a NextCloud solution easier to maintain

Regards,




 [cid:a57365a9-e3b3-44dc-9ae7-c5ba99592ccd]

Philippe Galiatsatos

Directeur technique



Téléphone : +33.(0)1.84.79.45.45

Mobile        : +33.(0)6.73.55.46.47

Email: ***@***.***


Abonnez vous à la lettre d'informations LSI<https://cdn.forms-content.sg-form.com/f974e319-4221-11ed-abe4-7669963d3d4f>



44 rue Brunel

Paris, 75017, France



www.lsiparis.tech



<http://linkedin.com/>  <http://twitter.com/>   <http://google.com/>



***@***.***?anonymous&ep=signature>               Réservez une heure pour me ***@***.***?anonymous&ep=signature>
________________________________
De : Ben ***@***.***>
Envoyé : mercredi 22 mars 2023 11:05
À : amidaware/tacticalrmm ***@***.***>
Cc : Philippe Galiatsatos ***@***.***>; Author ***@***.***>
Objet : Re: [amidaware/tacticalrmm] Install TRMM Agent on a synology DSM 7 (Discussion #1172)


Ran into the same issue on a RS422+<https://www.synology.com/de-de/products/RS422+>—are there any news regarding out-of-the-box Synology support?
